### ISSUE TITLE
LagFeatures not working properly?

### ISSUE THREAD

**Issue by nezirspanca** [feature-engine/feature_engine#814]
Hello all,

I have a problem using LagFeatures from Feature-engine. When I use the code below, I think that data_test should not contain NaN values ​​in the first lines, because the lag features will be created from the last lines of data_train. That's why I thought the term ".transform" was used. When I run this code, Lag Features are being created in data_train_transformed and data_test_transformed separately and in the same way.

I thought that LagFeatures of Feature-engine has this function, so using fit and transform, right?

# Lag features
lag_transformer = LagFeatures(
    variables = [
        "column1",
        "column2",
        "column3",
        "column4",
    periods=[1, 2, 3, 24, 168])
    
data_train_transformed = lag_transformer.fit_transform(data_train)

data_test_transformed = lag_transformer.transform(data_test)

Thank you!

**Comment by solegalli** [comment]
hi @nezirspanca 

To ensure the first rows in the test set do not contain NAN, you need to transform a test set that contains at least as many values in the past as you longer lag, respect to the first timepoint you have in the test set.

So if the bigger lag is 3 days, the test set needs to contain values at least 3 days before than the first day you want in the test set.

Say your train ends in september 30th and the test set begins in october 1st, to ensure you don't have nan in october 1st, the test set past to transform needs to start 3 days before, so september 28th.

Hope this helps
